packages/yoga/src/Theme/Provider/Provider.jsx

```diff
@@ -1,18 +1,50 @@
+import * as tokens from '@gympass/yoga-tokens';
+import PropTypes from 'prop-types';
 import React from 'react';
 import { ThemeProvider } from 'styled-components';
-import PropTypes from 'prop-types';
-import * as tokens from '@gympass/yoga-tokens';
 import yogaTheme from '../theme';
 
-const getTheme = ({ locale }) => {
+/**
+ * Gets resolved theme to inject in Yoga
+ *
+ * @param {{locale: string, theme: function}} themeParameters information about the theme
+ * @param {string} themeParameters.locale custom locale to get colors from (if it has a different scheme)
+ * @param {function} themeParameters.theme returns a custom theme object to replace existing tokens
+ * @returns {Object.<string, string | number>} theme object with tokens sent to styled-components' <ThemeProvider />
+ */
+const getTheme = ({ locale, theme }) => {
+  const isThemeInvalid = theme ? !('isFromThemeBuilder' in theme) : false;
```

Comment:
Seems like we have multiple branching only to validate the theme. What do you think of something like this:

```js
const validateAndRunBuilder = (builder, tokens) => {
  if (typeof builder === 'undefined') return false;
  if (!'isFromThemeBuilder' in theme) throw /* ... */;
  
  const customTheme = builder(tokens);
  if (typeof customTheme !== 'object') throw /* ... */;

  return customTheme;
}

const getTheme = ({ theme }) => {
  const token = tokens[locale] || tokens.default;
  const customTheme = validateAndRunBuilder(theme, token);

  /* ... */

  return customTheme ? yogaTheme(token, customTheme) : defaultTheme;
}
```